src/lsp/sql_parser/sql_parser.ml

```diff
@@ -12,64 +12,64 @@
 (**************************************************************************)
 open Cobol_common
 
-let supplier tokens = 
-  let state = ref tokens in 
+let supplier tokens =
+  let state = ref tokens in
   fun () ->
-    let token, locstart, locend = match !state with
-    | [] -> Grammar.EOF, Lexing.dummy_pos, Lexing.dummy_pos
-    | t:: s -> state :=s ; t
-  in
-  token, locstart, locend
+    let token, locstart, locend =
+      match !state with
+      | [] -> (Grammar.EOF, Lexing.dummy_pos, Lexing.dummy_pos)
+      | t :: s ->
+        state := s;
+        t
+    in
+    (token, locstart, locend)
 
 let init_checkpoint = Grammar.Incremental.main Lexing.dummy_pos
 
-let parse text =        
+let parse text =
   let module OM = Sql_overlay_manager in
   let tokens =
-    List.fold_left (fun (tokens, prev_right_limit) w' ->
-      let emit ~loc t =
-        let lstart, lend = OM.limits loc in
-        (match prev_right_limit with
-         | None -> ()
-         | Some l -> OM.link_limits l lstart);
-        (t, lstart, lend) :: tokens, Some lend
-      in
-      match w'.payload with
-      | Cobol_preproc.Text.TextWord tw ->
-        let tokenizer ~loc:_ lexbuf = Lexer.token lexbuf
-        in
-        let f (t': 'a with_loc) (tokens, prev_right_limit) =
-          let t = t'.payload in
-          let loc = t'.loc in
-          let lstart, lend = OM.limits (loc) in
-          (match prev_right_limit with
+    List.fold_left
+      (fun (tokens, prev_right_limit) w' ->
+        let emit ~loc t =
+          let lstart, lend = OM.limits loc in
+          ( match prev_right_limit with
           | None -> ()
-          | Some l -> OM.link_limits l lstart);
-          (t, lstart, lend) :: tokens, Some lend
-        in
-
-        let tw' : string with_loc = Srcloc.locfrom tw w' in
-
-        Tokenizing.fold_tokens ~tokenizer ~until: ((=) Grammar.EOF) ~f tw' 
-          (tokens, prev_right_limit)
-      | Cobol_preproc.Text.Separator s ->
-        let t = match s with
-          | ',' -> Grammar.COMMA
-          | ';' -> Grammar.SEMICOLON
-          | _ -> Grammar.SPECHAR(s)
+          | Some l -> OM.link_limits l lstart );
+          ((t, lstart, lend) :: tokens, Some lend)
         in
-        let lstart, lend = OM.limits w'.loc in
-        (match prev_right_limit with
-        | None -> ()
-        | Some l -> OM.link_limits l lstart);
-        (t, lstart, lend) :: tokens, Some lend
 
+        match w'.payload with
+        | Cobol_preproc.Text.TextWord tw ->
+          let tokenizer ~loc:_ lexbuf = Lexer.token lexbuf in
+          let f (t' : 'a with_loc) (tokens, prev_right_limit) =
+            let t = t'.payload in
+            let loc = t'.loc in
+            let lstart, lend = OM.limits loc in
```

Comment:
Just add the `acc` (= `tokens, prev_right_limit`) argument to `emit` (to be moved right before `let tokens =` above), and use `emit` here as well.